src/Shouldly.Tests/ShouldThrowAsync/FuncOfTaskScenarioAsync.cs

```diff
@@ -157,5 +158,27 @@ but threw
                     System.DivideByZeroException");
             }
         }
+
+        [Fact] // Issue 818
+        public async Task ShouldThrowAssertException_ExceptionTypeIsException()
+        {
+            try
+            {
+                Func<Task> doSomething = () => Task.CompletedTask;
+                await Should.ThrowAsync<Exception>(async () => await doSomething());
+            }
+            catch (ShouldAssertException e)
```

Comment:
I think it would be better to use `catch (Exception e)` here to catch unexpected errors and get an appropriate error message from the `ShouldBeOfType` assertion below.

Reply:
Agreed and updated, it now matches the tests above, I think I had it leftover as ShouldAssertException from my initial testing.